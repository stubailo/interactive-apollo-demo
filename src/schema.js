import { graphql } from 'graphql';
import { makeExecutableSchema } from 'graphql-tools';

const typeDefs = `
  type Query {
    benefits(first: Int!): [Benefit]
  }

  type Benefit {
    title: String!
    content: String!
  }
`;

const resolvers = {
  Query: {
    benefits(root, { first }) {
      if (first < 1 || first > data.length) {
        throw new Error('first argument out of range.');
      }

      return data.slice(0, first);
    }
  }
}

const schema = makeExecutableSchema({ typeDefs, resolvers });

export function fetchQuery({ query, variables }) {
  return graphql(schema, query, null, null, variables);
}

const data = [
  {
    title: 'ASDF',
    content: 'ASDF [link](example.com) other content',
  },
];
