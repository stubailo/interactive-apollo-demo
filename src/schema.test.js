import { fetchQuery } from './schema';

it('runs a basic query', () => {
  return fetchQuery({
    query: `{ benefits(first: 1) { title } }`,
  }).then(( result ) => {
    expect(result).toMatchSnapshot();
  });
});
