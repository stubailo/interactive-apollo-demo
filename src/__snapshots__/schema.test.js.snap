exports[`test runs a basic query 1`] = `
Object {
  "data": Object {
    "benefits": Array [
      Object {
        "title": "ASDF",
      },
    ],
  },
}
`;
